apps/synapse-oauth-signin/src/OAuth2Form.tsx

```diff
@@ -312,7 +313,7 @@ export function OAuth2Form() {
       }}
     >
       <div style={{ marginTop: '50px' }}>
-        {showPendingRedirectUI && (
+        {pendingRedirectURL !== undefined && (
```

Comment:
Suggest still creating a variable derived from the new state for clarity, e.g.

`const showPendingRedirectUi = pendingRedirectURL !== undefined`